src/main/java/uk/gov/hmcts/darts/cases/service/impl/CloseOldCasesProcessorImpl.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.hmcts.darts.cases.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.darts.cases.service.CloseOldCasesProcessor;
+import uk.gov.hmcts.darts.common.entity.CourtCaseEntity;
+import uk.gov.hmcts.darts.common.entity.EventEntity;
+import uk.gov.hmcts.darts.common.entity.HearingEntity;
+import uk.gov.hmcts.darts.common.entity.MediaEntity;
+import uk.gov.hmcts.darts.common.repository.CaseRepository;
+import uk.gov.hmcts.darts.common.repository.EventRepository;
+import uk.gov.hmcts.darts.common.repository.MediaRepository;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+import static java.lang.Boolean.TRUE;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class CloseOldCasesProcessorImpl implements CloseOldCasesProcessor {
+    private final CaseRepository caseRepository;
+
+    private final EventRepository eventRepository;
+
+    private final MediaRepository mediaRepository;
+
+
+    @Transactional
+    @Override
+    public void closeCases() {
+        List<CourtCaseEntity> courtCaseEntityList = caseRepository.findOpenCasesToClose(OffsetDateTime.now().minusYears(6));
+
+        courtCaseEntityList.forEach(this::closeCase);
+    }
+
+    private void closeCase(CourtCaseEntity courtCase) {
+        List<EventEntity> eventList = new ArrayList<>();
+        for (HearingEntity hearingEntity: courtCase.getHearings()) {
+            eventList.addAll(hearingEntity.getEventList());
+        }
+        if (eventList.size() > 1) {
+            eventList.sort(Comparator.comparing(EventEntity::getCreatedDateTime).reversed());
+            //find latest closed event
+            Optional<EventEntity> closedEvent =
+                eventList.stream().filter(eventEntity -> eventEntity.getEventType().getEventName().equals("Case closed")).findFirst();
```

Comment:
Hardcoding this to "case closed" doesn't seem like the best thing to do. Firstly, this is an XHIBIT even type, and so they could change the name of that event at any time. Secondly, CPP has their own case closed event called "Archive case", and this logic wouldn't work for that scenario. Happy to talk this through and try and come up with a solution that is future proof if you need.